lib/Video/Adapter/Ffmpeg.php

```diff
@@ -262,14 +262,16 @@ public function getDuration()
         $output = $this->getVideoInfo();
 
         // get total video duration
-        preg_match("/Duration: ([0-9:\.]+),/", $output, $matches);
-        $durationRaw = $matches[1];
-        $durationParts = explode(':', $durationRaw);
+        $result = preg_match("/\s+Duration: ((\d\d):(\d\d):(\d\d)\.(\d+))/", $output, $matches);
+
+        if (false !== $result && 0 !== $result && count($matches) === 6) {
+            // calculate duration in seconds
+            $duration = ((int)$matches[2] * 3600) + ((int)$matches[3] * 60) + (float)$matches[4];
```

Comment:
So the milliseconds should be in the return value again:
```suggestion
        $result = preg_match("/\s+Duration: (\d\d):(\d\d):(\d\d\.\d+)/", $output, $matches);

        if ($result) {
            // calculate duration in seconds
            $duration = ((int)$matches[1] * 3600) + ((int)$matches[2] * 60) + (float)$matches[3];
```

Reply:
it said already "calculate duration in seconds", thats why the milliseconds got ignored.

Reply:
As you can see the 3rd value is a float. The milliseconds were there. This has worked before. The return is in seconds, the milliseconds are only after the decimal point.

Reply:
@blankse alright, added your suggestion